### ISSUE TITLE
Another layout throws "out of range" error

### ISSUE THREAD

**Issue by a1mery** [SharePoint/sp-dev-modernization#426]
#### Category

[*] Bug
[ ] Enhancement

#### Problem Area

[ ] Page Transformation: Error during the setup/use of the Page Transformation UI solution (did you check our [troubleshooting guide](https://github.com/SharePoint/sp-dev-modernization/blob/dev/Solutions/PageTransformationUI/docs/troubleshootingguide.md)?)
[ ] Page Transformation: Error during the use of page transformation from PnP PowerShell
[*] Page Transformation: Error during the use of page transformation from .Net
[ ] Page Transformation: Page is not looking correct after transformation
[ ] Modernization Scanner: something went wrong...

#### Expected or Desired Behavior

A classic WebPart page with the following layout : Right Column, Header, Footer, Top Row, 3 Columns is correctly transformed into a modern client-side page. The target classic page has:

- 1 WebPart in the Header
- 1 WebPart in the Center Column
- 2 WebParts in the Center-Right Column

or

- 2 WebParts in the Header
- 1 WebPart in the Right Column
- 2 WebParts in the Center-Left Column
- 2 WebParts in the Center Column

#### Observed Behavior

The page transformation engine throws the following error :

> Page ClassicPage.aspx could not be transformed: Index was out of range. Must be non-negative and less than the size of the collection.
Parameter name: index

#### Steps to Reproduce

1. On a modern team site (no Office 365 group), create a WebPart page inside the Site Pages library with the following page layout : Right Column, Header, Footer, Top Row, 3 Column.
2. Insert a WebPart in the following zones:
- 1 WebPart in the Header
- 1 WebPart in the Center Column
- 2 WebParts in the Center-Right Column

or

- 2 WebParts in the Header
- 1 WebPart in the Right Column
- 2 WebParts in the Center-Left Column
- 2 WebParts in the Center Column

3. Run the page transformation engine (page transformation from .Net using SharePointPnP.Modernization.Framework.dll version 1.0.2001).

#### Additional informations

This issue might be linked with this one : #371 
We can reproduce the same behaviour with the following page layout : LeftColumn, Header, Footer, Top Row, 3 Column

Thanks for your help

**Comment by jansenbe** [comment]
Thanks for logging this @a1mery , I'm able to repro and will fix this for the upcoming February release

**Comment by jansenbe** [comment]
@a1mery : problem has been fixed. The issue was that we normalized row numbers which resulted in a row number shift (web parts in 3 column row ended up in the above row which just had 1 column, causing the error). The solution is to not do row number normalizing for web part pages as it makes no sense there, this only is needed for wiki pages.

The fix will be part of the February release which will come out either tomorrow or Monday next week.
